src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -55,8 +195,6 @@ public override TypedExpression Transform(TypedExpression ex)
 
             if (types.Any() && ex.Expression is QsExpressionKind<TypedExpression, Identifier, ResolvedType>.CallLikeExpression call)
```

Comment:
What is there are type parameter resolutions but the expression is not a call-like expression?

Reply:
I didn't quite understand your question here, but you should know that this section was entirely rewritten in #206 .